src/Auth/OAuth.php

```diff
@@ -24,20 +24,27 @@ abstract class OAuth
      */
     protected $state;
 
+    /**
+     * @var array
+     */
+    protected $userScopes;
+
     /**
      * OAuth constructor.
      *
      * @param string $appId
      * @param string $appSecret
      * @param string $callback
      * @param array  $state
+     * @param array $userScopes
      */
-    public function __construct(string $appId, string $appSecret, string $callback, $state = [])
+    public function __construct(string $appId, string $appSecret, string $callback, $state = [], $userScopes = [])
```

Comment:
When passing scopes directly to the protected $userScopes attribute we skip the duplication check. Maybe we can use the addScope internally from inside the object constructor.

Also I think that in the context of this class the 'userScopes' variable can be named only 'scopes'.

Reply:
could you please elaborate this? I dont think I understood

Reply:
instead of this assignment:
```
$this->userScopes = $userScopes;
```

we should have something like this:
```
foreach($scopes as $scope) {
    $this->addScope($scope);
}
```

Reply:
okay noted.